gestalt-asset-core/src/main/java/org/terasology/gestalt/assets/AssetType.java

```diff
@@ -288,15 +269,45 @@ synchronized void registerAsset(Asset<U> asset, DisposalHook disposer) {
         if (closed) {
             throw new IllegalStateException("Cannot create asset for disposed asset type: " + assetClass);
         } else {
-            if (asset.getUrn().isInstance()) {
-                instanceAssets.put(asset.getUrn(), new WeakReference<>(assetClass.cast(asset)));
-            } else {
-                loadedAssets.put(asset.getUrn(), assetClass.cast(asset));
+            if(!asset.getUrn().isInstance()) {
```

Comment:
This is just a general style comment - should be a space after if, happens in a number of changes in this PR. Needs a style cleanup.